desc/profiles.py

```diff
@@ -422,13 +446,10 @@ def compute(self, params=None, grid=None, dr=0, dt=0, dz=0):
             values of the profile or its derivative at the points specified.
 
         """
-        if params is None:
-            params = [None] * len(self._profiles)
-        if isinstance(params, dict):
-            params = [params]
+        params = self._parse_params(params)
         f = 0
         for i, profile in enumerate(self._profiles):
-            f += profile.compute(params[i % len(params)], grid, dr, dt, dz)
+            f += profile.compute(params[i], grid, dr, dt, dz)
```

Comment:
why did we need to modulo this before, and why do we not need to do it now

Reply:
the new getter/setter and _parse param methods should ensure that params is always the right length